controllers/tasks.js

```diff
@@ -0,0 +1,49 @@
+const Task = require('../models/Task')
+const asyncWrapper = require('../middleware/async')
+const {createCustomError, CustomAPIError} = require('../errors/custom-error')
+
+const getAllTasks = asyncWrapper(async(req,res)=> {
+    const tasks = await Task.find({})
+    res.status(200).json({tasks})
+})
+
+const createTask = asyncWrapper(async (req,res)=>{
+    const task = await Task.create(req.body)
+    res.status(201).json({task});
+    
+})
+
+const getTask = asyncWrapper(async(req,res,next)=>{
+    const {id:taskID} = req.params
+    const task = await Task.findOne({_id:taskID})
+    if (!task){
+        return next(createCustomError(`No task with id :${taskID}`,404))
+    }
+    res.status(200).json({task});
+})
+
+const deleteTask = asyncWrapper(async(req,res)=>{
+    const {id:taskID} = req.params
+    const task = await Task.findOneAndDelete({_id:taskID})
```

Comment:
Notice that you use directly the user's paramets on your own DB's & infrustructures. That's not secure, and you should always do some validation on the user input. Never trust the user :)

Reply:
פתחת את הפרונט (וראית את שמות המשימות)? כי אפשר לראות שם שהיוזר בעצם מקליק על משימה בשביל למחוק אותה. אז כמה טעויות יכולות לקרות כבר משם?
מה שלא הבנתי (ואולי זה מתרחש בפרונט) זה איך עצם זה שהיוזר לוחץ על הלחצן דליט גורם לכך שהיוזר פתאום יודע איזה ID לספק לבקשת DELETE כדי שהיא באמת תקרה?
מודה שאפילו לא פתחתי את הפרונט